### PR TITLE
Feature/validator api client endpoints

### PR DESCRIPTION
**Pull request by jstuczyn**
It depends on https://github.com/nymtech/nym/pull/1019 and thus remains as a draft until the aforementioned PR is merged.

### CODE REVIEW

File: nym-wallet/src-tauri/Cargo.toml [hunk 1]
```diff
@@ -49,7 +49,7 @@ features = ["ts-rs"]
 
 [dev-dependencies.validator-client]
 path = "../../common/client-libs/validator-client"
-features = ["ts-rs"]
+features = ["typescript-types"]
```
Comment (mmsinclair) [inline nym-wallet/src-tauri/Cargo.toml:52]: Is there still a GitHub Action that builds the types? And would this affect it?

File: validator-api/validator-api-requests/src/models.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[cfg_attr(feature = "ts-rs", derive(ts_rs::TS))]
```
Comment (mmsinclair) [inline validator-api/validator-api-requests/src/models.rs:7]: Is the feature = `ts-rs` correct still?

**Review by mmsinclair — APPROVED** [review]
Looks good - only a couple of questions about the typescript feature flag

File: nym-wallet/src-tauri/Cargo.toml (hunk 1 above)
Comment (jstuczyn) [reply nym-wallet/src-tauri/Cargo.toml:52]: The type building is done via simple `cargo test` command which is still done as part of our CI pipeline and hence it should be unaffected.

File: validator-api/validator-api-requests/src/models.rs (hunk 2 above)
Comment (jstuczyn) [reply validator-api/validator-api-requests/src/models.rs:7]: Correct. If you notice the `typescript-types` was defined for the client crate and this refers to the `validator-api-requests` crate. This crate was introduced to remove duplication of type definitions. And as for the new "feature", it was defined due to name clash of `ts-rs`

File: common/client-libs/validator-client/src/validator_api/mod.rs [hunk 3]
```diff
@@ -1,16 +1,24 @@
-// Copyright 2021 - Nym Technologies SA <contact@nymtech.net>
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
```
Comment (neacsu) [inline common/client-libs/validator-client/src/validator_api/mod.rs]: The correct copyright would be 2021-2022 I think, but if that's not how we did it in the rest of the codebase then I guess it doesn't matter that much. Or we could start now :?

**Review by neacsu — APPROVED** [review]

File: common/client-libs/validator-client/src/validator_api/mod.rs (hunk 3 above)
Comment (jstuczyn) [reply common/client-libs/validator-client/src/validator_api/mod.rs]: Good question. @mmsinclair or @futurechimp any opinions on that?

Comment (mmsinclair) [reply common/client-libs/validator-client/src/validator_api/mod.rs]: My preference would be `2021-2022`

Comment (jstuczyn) [reply common/client-libs/validator-client/src/validator_api/mod.rs]: As discussed on keybase, @mmsinclair aggress with @neacsu and I'm gonna update that header accordingly.

**Pull request merged**